lib/Runtime/Base/ThreadContext.h

```diff
@@ -423,7 +423,28 @@ class ThreadContext sealed :
 #endif
 #endif
 
+public:
+    Js::PropertyRecord const * GetEmptyStringPropertyRecord()
+    {
+        if (!emptyStringPropertyRecord)
+        {
+            emptyStringPropertyRecord = propertyMap->LookupWithKey(Js::HashedCharacterBuffer<char16>(_u(""), 0));
+            if (emptyStringPropertyRecord == nullptr)
+            {
+                emptyStringPropertyRecord = this->UncheckedAddPropertyId(_u(""), 0);
```

Comment:
The ThreadContext is heap allocated, but the property record you're allocating here is Recycler allocated so it can cause the PropertyRecord to get collected prematurely and can cause GetEmptyStringPropertyRecord to return a pointer to a record that's already been collected.

I think this is a bug with the existing code that can cause functional issues- for now, the simplest thing to do is call UncheckedAddPropertyId(_u(""), 0, true /* bind */) so that we keep the property record alive for the duration of the thread context.

Reply:
Good point. Thanks